electron/gog/games.ts

```diff
@@ -124,75 +129,79 @@ class GOGGame extends Game {
-          executable: '',
-          install_path: join(path, gameInfo.folder_name),
-          install_size: prettyBytes(installInfo.manifest.disk_size),
-          is_dlc: false,
-          version: additionalInfo
-            ? additionalInfo.version
-            : installInfo.game.version,
-          appName: this.appName,
-          installedWithDLCs: installDlcs,
-          language: installLanguage,
-          versionEtag: isLinuxNative ? '' : installInfo.manifest.versionEtag,
-          buildId: isLinuxNative ? '' : installInfo.game.buildId
-        }
-        const array: Array<InstalledInfo> =
-          (installedGamesStore.get('installed') as Array<InstalledInfo>) || []
-        array.push(installedData)
-        installedGamesStore.set('installed', array)
-        GOGLibrary.get().refreshInstalled()
-        if (isWindows) {
-          await setup(this.appName)
-        }
-        return { status: 'done' }
-      })
-      .catch(() => {
-        logInfo('Installaton canceled', LogPrefix.Gog)
-        return { status: 'error' }
-      })
+    const logPath = join(heroicGamesConfigPath, this.appName + '.log')
+
+    const commandParts = [
+      'download',
+      this.appName,
+      '--platform',
+      installPlatform,
+      `--path=${path}`,
+      `--token=${credentials.access_token}`,
+      withDlcs,
+      `--lang=${installLanguage}`,
+      workers
+    ]
+    const command = getGogdlCommand(commandParts)
+
+    logInfo([`Installing ${this.appName} with:`, command], LogPrefix.Gog)
+
+    const res = await runGogdlCommand(commandParts, logPath)
+
+    if (res.error) {
+      logError(
+        ['Failed to install', `${this.appName}:`, res.error],
+        LogPrefix.Gog
+      )
+      return { status: 'error' }
+    }
+
+    // Installation succeded
+    // Save new game info to installed games store
+    const installInfo = await this.getInstallInfo()
+    const gameInfo = GOGLibrary.get().getGameInfo(this.appName)
+    const isLinuxNative = installPlatform == 'linux'
+    const additionalInfo = isLinuxNative
+      ? await GOGLibrary.getLinuxInstallerInfo(this.appName)
+      : null
+    const installedData: InstalledInfo = {
+      platform: installPlatform,
+      executable: '',
+      install_path: join(path, gameInfo.folder_name),
+      install_size: prettyBytes(installInfo.manifest.disk_size),
+      is_dlc: false,
+      version: additionalInfo
+        ? additionalInfo.version
+        : installInfo.game.version,
+      appName: this.appName,
+      installedWithDLCs: installDlcs,
+      language: installLanguage,
+      versionEtag: isLinuxNative ? '' : installInfo.manifest.versionEtag,
+      buildId: isLinuxNative ? '' : installInfo.game.buildId
+    }
+    const array: Array<InstalledInfo> =
+      (installedGamesStore.get('installed') as Array<InstalledInfo>) || []
+    array.push(installedData)
+    installedGamesStore.set('installed', array)
+    GOGLibrary.get().refreshInstalled()
```

Comment:
About bug you mentioned. This might be a race condition we are looking for, since refreshInstalled reads from the same config. Maybe a function to update installed games directly in memory (instead of reading from the same file again) will solve the issue.

Reply:
Maybe we could do it like this from now on. Perhaps on another PR:
After refreshing the library on the backend, we send an event to the frontend to refresh the state.
This way we could solve the racing conditions.